platforms/chibios/drivers/audio_pwm_hardware.c

```diff
@@ -44,8 +59,12 @@ void channel_1_set_frequency(float freq) {
 
     if (freq <= 0.0) {
         // a pause/rest has freq=0
+        channel_1_stop();
         return;
     }
+    if (channel_1_stopped) {
+        channel_1_start();
+    }
```

Comment:
This function is called from the timer interrupt handler; calling `pwmStop()` or `pwmStart()` in this state is not valid.

What could be valid is:
```suggestion
        chSysLockFromISR();
        pwmChangePeriodI(&AUDIO_PWM_DRIVER, 2);
        pwmEnableChannelI(&AUDIO_PWM_DRIVER, AUDIO_PWM_CHANNEL - 1, 0);
        chSysUnlockFromISR();
        return;
    }
```